enclave-runner/src/usercalls/mod.rs

```diff
@@ -506,7 +506,8 @@ struct IOHandlerInput<'tcs> {
 }
 
 struct PendingEvents {
-    counts: [u32; Self::EV_MAX],
+    counts: [Semaphore; Self::EV_MAX],
```

Comment:
Can you add a comment here briefly explaining the meaning of this field? I don't think an array of semaphores is a very common construct (which is to say I have never seen it before)

Reply:
It's effectively the same thing as before (a bunch of counters for all possible event sets) but with the added benefit of being able to wait until a counter is non zero (through `Semaphore::acquire`). Although you may be right that this could be confusing for readers, so I'll add a comment.